### PR TITLE
Mradev 923 refactor resizer

### PR DESCRIPTION
**Pull request by cstetson**
This is a significant refactor of the resizer:
- Moved all the transform functions out of PhotoResizer.java and into a new class PhotoTransformer.java
- Moved all file and network IO out of PhotoResizer.java and into a new class PhotoIO.java (this will give us a better base to change the interaction with Uploader)
- Updated JavaDocs to reflect changes

### CODE REVIEW

**Comment by SirNexus** [comment]
Built and tested it. Looks good!

File: app/src/main/java/com/nginx/image/core/PhotoTransformer.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            case 7: // PI/2 and Flip
+                t.scale(-1.0, 1.0);
+                t.translate(-info.height, 0);
+                t.translate(0, info.width);
+                t.rotate(  3 * Math.PI / 2);
+                break;
+            case 8: // PI / 2
+                t.translate(0, info.width);
+                t.rotate(  3 * Math.PI / 2);
+                break;
```
Comment (cpretzer) [inline app/src/main/java/com/nginx/image/core/PhotoTransformer.java]: It's good practice to have a default case in a switch statement. Can we be 100% certain that the value will always be one of those covered by the case statements?

File: .dockerignore [hunk 2]
```diff
@@ -11,5 +11,4 @@ app/mozilla/
 app/net/
 app/newrelic/
 app/org/
-app/target/
-deploy
```
Comment (cpretzer) [inline .dockerignore]: I originally added this line in order to improve build performance. Do the changes in this refactor require that the deploy directory is copied to the image? If not, including this line saves time and decreases the image size.

File: app/src/main/java/com/nginx/image/core/PhotoTransformer.java (hunk 1 above)
Comment (eduardborges) [reply app/src/main/java/com/nginx/image/core/PhotoTransformer.java]: I agree with @cpretzer . We should have a default case.

Comment (cstetson) [reply app/src/main/java/com/nginx/image/core/PhotoTransformer.java]: good call, and easily done

File: .dockerignore (hunk 2 above)
Comment (cstetson) [reply .dockerignore]: no, it was during the rebase that I chose one way rather than the other, not with a particular reason in mind

**Pull request merged**